xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -280,6 +282,24 @@ func (cib *clusterImplBalancer) UpdateSubConnState(sc balancer.SubConn, s balanc
 		return
 	}
 
+	// Trigger re-resolution when a SubConn turns transient failure. This is
+	// necessary for the LogicalDNS in cluster_resolver policy to re-resolve.
+	//
+	// Note that this happens not only for the addresses from DNS, but also for
+	// EDS (cluster_impl doesn't know if it's DNS or EDS, only the parent
+	// knows). The parent priority policy is configured to ignore re-resolution
+	// signal from the EDS children.
+	//
+	// Another option is to add a boolean to cluster_impl's config, and only
+	// trigger re-resolution if that's true. But this requires changing the
+	// balancer config, and is not consistent for all languages.
+	//
+	// The final goal is to move re-resolution into the leaf policies
+	// (round_robin), so that this won't be necessary.
```

Comment:
This all sounds fine, but I don't think we need a comment for it.

Reply:
Removed.